src/dashd.cpp

```diff
@@ -142,29 +143,21 @@ bool AppInit(int argc, char* argv[])
             fprintf(stderr, "Error: There is no RPC client functionality in dashd anymore. Use the dash-cli utility instead.\n");
             exit(EXIT_FAILURE);
         }
-#ifndef WIN32
         if (GetBoolArg("-daemon", false))
         {
+#if HAVE_DECL_DAEMON
             fprintf(stdout, "Dash Core server starting\n");
 
             // Daemonize
-            pid_t pid = fork();
-            if (pid < 0)
-            {
-                fprintf(stderr, "Error: fork() returned %d errno %d\n", pid, errno);
+            if (daemon(1, 0)) { // don't chdir (1), do close FDs (0)
+                fprintf(stderr, "Error: daemon() failed: %s\n", strerror(errno));
                 return false;
             }
-            if (pid > 0) // Parent process, pid is child process id
-            {
-                return true;
-            }
-            // Child process falls through to rest of initialization
-
-            pid_t sid = setsid();
-            if (sid < 0)
-                fprintf(stderr, "Error: setsid() returned %d errno %d\n", sid, errno);
+#else
+            fprintf(stderr, "Error: -daemon is not supported on this operating system\n");
```

Comment:
Now I'm getting this error on Ubuntu 17. I'm trying to understand details now,  but it looks like a problem

Reply:
Make sure to `make clean && ./autogen.sh`

Reply:
Yes, I'm sorry, it's my error,  I've missed autogen.sh.